onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -251,11 +257,16 @@ void CopyDataToTensor(PyArrayObject* darray, int npy_type, std::unique_ptr<Tenso
     if (!IAllocator::CalcMemSizeForArray(p_tensor->DataType()->Size(), p_tensor->Shape().Size(), &len)) {
       throw std::runtime_error("length overflow");
     }
-    memcpy(buffer, PyArray_DATA(darray), len);
+    mem_cpy_to_device(buffer, PyArray_DATA(darray), len);
   }
 }
 
-std::unique_ptr<Tensor> CreateTensor(const AllocatorPtr& alloc, const std::string& name_input, PyArrayObject* pyObject) {
+// Setting `use_numpy_data_memory` to `true` will ensure that the underlying numpy array buffer is directly used
```

Comment:
>ensure [](start = 50, length = 6)

I am actually reading it in a different way. Previously it would always use the buffer directly if it can but now an additional boolean is required.

Reply:
yes, there is an additional boolean that basically gives the user ***the option*** of not using the numpy data buffer as the backing data buffer for the Tensor.  However, the default behavior is still to use the numpy buffer directly (no copy approach). It gives the caller an option to trigger the "copy infrastructure"  and thus de-couple the life cycle of the Tensor from the numpy array (should the caller wish to take that approach). This is an internal API and personally I feel giving the caller the flexibility is not necessarily a bad thing. Do you have any specific concerns ?